src/utils/rawconfig.py

```diff
@@ -0,0 +1,62 @@
+# System Packages
+from pathlib import Path
+from typing import List, Optional
+
+# External Packages
+from pydantic import BaseModel
+
+# Internal Packages
+from src.utils.helpers import to_snake_case_from_dash
+
+class ConfigBaseModel(BaseModel):
+    class Config:
+        alias_generator = to_snake_case_from_dash
+        allow_population_by_field_name = True
+
+class SearchConfigTest(ConfigBaseModel):
+    input_files: Optional[List[str]]
+    input_filter: Optional[str]
+    embeddings_file: Optional[Path]
+
+class TextSearchConfigTest(ConfigBaseModel):
+    compressed_jsonl: Optional[Path]
+    input_files: Optional[List[str]]
+    input_filter: Optional[str]
+    embeddings_file: Optional[Path]
+
+class ImageSearchConfigTest(ConfigBaseModel):
+    use_xmp_metadata: Optional[str]
+    batch_size: Optional[int]
+    input_directory: Optional[List[str]]
+    input_filter: Optional[str]
+    embeddings_file: Optional[Path]
+
+class ContentType(ConfigBaseModel):
+    org: Optional[TextSearchConfigTest]
+    ledger: Optional[TextSearchConfigTest]
+    image: Optional[ImageSearchConfigTest]
+    music: Optional[TextSearchConfigTest]
+
+class AsymmetricConfig(ConfigBaseModel):
+    encoder: Optional[str]
+    cross_encoder: Optional[str]
+
+class ImageSearchTypeConfig(ConfigBaseModel):
+    encoder: Optional[str]
+
+class SearchTypeConfigTest(ConfigBaseModel):
+    asymmetric: Optional[AsymmetricConfig]
+    image: Optional[ImageSearchTypeConfig]
+
+class ProcessorConversationConfig(ConfigBaseModel):
+    open_api_key: Optional[str]
```

Comment:
Shouldn't `open_api_key`, `conversation_logfile` and `conversation_history` all be required field in the `ProcessorConversationConfig`?

Reply:
Yes, but changing this will require refactoring the code a bit. I would suggest doing this in a separate CR, so that this one can be limited to creating the config UI and integrating the most minimal addition of BaseModel types. Opening #12 to track that work.